main.go

```diff
@@ -75,24 +74,28 @@ func main() {
 	}
 	// Used by go-bluetooth, and we use this to set how much it logs. Not for this project.
 	logrus.SetLevel(logrus.InfoLevel)
+
+	// define environment
+	var env gateways.Environment
+	if ENV == "prod" {
+		env = gateways.Production
+	} else {
+		env = gateways.Development
+	}
 	// temporary for us, for release want info level - todo make configurable via cli?
 	zerolog.SetGlobalLevel(zerolog.DebugLevel)
 
 	logger.Info().Msgf("Starting DIMO Edge Network, with log level: %s", zerolog.GlobalLevel())
+	// setup datasender here so we can send errors to it
+	ds := network.NewDataSender(unitID, *ethAddr, logger, nil)
```

Comment:
move the `if ethAddr == nil` check lines below before this, since *ethAddr could nil pointer exception

Reply:
oh, good catch